src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -520,6 +520,12 @@ private static FunctionMetadata ParseFunctionMetadata(string functionName, JObje
                 functionMetadata.IsExcluded = (bool)value;
             }
 
+            if (configMetadata.TryGetValue("autoPackageRestoreEnabled", StringComparison.OrdinalIgnoreCase, out value) &&
+                value.Type == JTokenType.Boolean)
+            {
+                functionMetadata.IsAutoPackageRestoreEnabled = (bool)value;
```

Comment:
I don't think we need the flag at all - users will expect this to be the default I think. It's easy enough to add a flag in the future if we get users asking to be able to turn it off, but I'm not sure why they would if it is working correctly/efficiently. Or, if we decide we NEED to have a way to disable this, I'd go with a single host level setting, defaulting to true, allowing users to turn this off. Though I don't see why people would need to turn it off. For functions not needing this (i.e. no packages) the implementation should be extremely efficient so there is really no impact, right?

The argument about "stable" functions applies to a lot of other things we set up as well, file watchers, etc. Those are only really needed at development time, just like nuget restore. We don't have knobs enabling people to disable those other development features either.